Add Documentation after 0.3.0 release (#94)
AbsaOSS/hermes: _docs/0.3/components/dataset-comparison.md

---
layout: docs
title: Compontent - Dataset Comparison
version: '0.3'
categories:
    - '0.3'
    - components
redirect_from: /docs/components/dataset-comparison
---

#### Table of contents

- [Table of contents](#table-of-contents)
- [Features](#features)
- [Features for spark-job](#features-for-spark-job)
- [Constraints](#constraints)
- [Concepts](#concepts)
  - [Schema Alignment](#schema-alignment)

>   - [Provided Schema](#provided-schema)

Reviewer: Capitalization seems inconsistent. (_Starting with a capital letter only and be done with it_ or _All Capitals with the Exception of Prepositions, Conjunctions, and Articles_? [Help](https://www.webucator.com/how-to/how-capitalize-headings-titles.cfm))

Author: Fixed

#### Features

- compares 2 datasets and provides a delta file of sorts together with some metrics and metadata

> - if keys are provided, the tool is specific in its output, providing paths to differences and showing the diff data.

Reviewer 2: As suggested above. Are they really keys - multiple ones? Or fields of one key?

- schema can be supplied for selective comparison. This will only compare the fields found in the schema

#### Features for spark-job

> - can load any data type/format that spark is able to load. It might be needed to add libraries to the class path

Reviewer 2: If including the second sentence at all, then: 
```suggestion
- can load any data type/format that spark is able to load. For some of the formats to be supported additional  libraries might need to be added to the class path
```

> - input data referential or new (being tested) can have different formats

Reviewer 2: Not sure I understand what you trying to say here. That the two data sets to compare can originate from different formats?

- format of the diff file can be configured as well (default is `parquet`)

> - regardless of fail or pass of the data, a metrics file called `_METRICS` is wirtten to the destination

Reviewer 2: ```suggestion
- regardless of fail or pass status of the data comparison, a metrics file called `_METRICS` is wirtten to the destination
```

#### Constraints

> - referential schema must be subset of the new (being verified) data

Reviewer: ```suggestion
- referential schema must be a subset of the new (being verified) data
```

> - without provided key, the diff file does a comparison of a row as a whole. With keys, it compares precise data and shows paths to differences

Reviewer: ```suggestion
- without a provided key, the diff file does a comparison of a row as a whole. With keys, it compares precise data and shows paths to differences
```

> - without provided key, the diff file does a comparison of a row as a whole. With keys, it compares precise data and shows paths to differences

Reviewer 2: Just not to be omitted. The (potential) misleading  of _keys_ vs _fields of the key_ applies here too.

#### Concepts

##### Schema Alignment

> Dataset comparison takes the referential data and tries to aling the new (being verified) data to the schema of the referential data. This means both sorting the columns and only selecting columns that are in the referential data.

Reviewer: ```suggestion
Dataset comparison takes the referential data and tries to align the new (being verified) data to the schema of the referential data. This means both sorting the columns and only selecting columns that are present in the referential data.
```

- [Features for spark-job](#features-for-spark-job)
- [Constraints](#constraints)
- [Concepts](#concepts)
  - [Schema alignment](#schema-alignment)
  - [Provided schema](#provided-schema)

#### Features

- compares two datasets and provides a delta file of sorts together with some metrics and metadata
- if keys are provided, the tool is specific in its output, providing paths to differences and showing the diff data.
- schema can be supplied for selective comparison. This will only compare the fields found in the schema

#### Features for spark-job

> - can load any data type/format that Apache Spark is able to load. For some of the formats to be supported additional  libraries might need to be added to the class path

Reviewer: ```suggestion
- can load any data type/format that Apache Spark is able to load. For some of the formats to be supported, additional libraries might need to be added to the class path
```

- input data referential or new (being tested) can have different formats
- format of the diff file can be configured as well (default is `parquet`)
- regardless of fail or pass status of the data comparison, a metrics file called `_METRICS` is wirtten to the destination

#### Constraints

- referential schema must be a subset of the new (being verified) data
- without a provided key, the diff file makes a comparison of a row as a whole. With keys, it compares precise data and shows paths to differences

#### Concepts

##### Schema alignment

Dataset comparison takes the referential data and tries to align the new (being verified) data to the schema of the referential data. This means both sorting the columns and only selecting columns that are present in the referential data.

##### Provided schema

> If a schema is provided for a dataset comparison, then the `Schema Alignment` is done against this provided schema for both referential and new data.

Reviewer: ```suggestion
If a schema is provided for a dataset comparison, then _schema alignment_ is done against this provided schema for both referential and new data.
```
